src/pint/models/pulsar_binary.py

```diff
@@ -185,7 +187,31 @@ def apply_units(self):
             bparObj.value = bparObj.value * u.Unit(bparObj.units)
 
     def update_binary_object(self, toas=None, acc_delay=None):
-        """Update binary object instance for this set of parameters/toas."""
+        """Update stand alone binary's parameters and toas from PINT-facing object.
+
+        This function passes the PINT-facing object's parameter values and TOAs
+        to the stand-alone binary object. If the TOAs are not provided, it only
+        updates the parameters not the TOAs.
```

Comment:
I think this is unclear. If I understand correctly, not providing TOAs means that the parameters are updated, but it is not the actual TOAs that are retained unchanged - this object retains barycentered TOAs internally, so that if you change the sky position (for example), the retained values are no longer the values you would obtain if providing the same TOAs. This is a very surprising behaviour that will affect anyone using this function with the default values, and it should be clearly stated in the documentation.

Reply:
I added a warning.